neuro_san/internals/messages/agent_framework_message.py

```diff
@@ -9,26 +9,37 @@
 # neuro-san SDK Software in commercial settings.
 #
 # END COPYRIGHT
+from __future__ import annotations
+
 from typing import Any
 from typing import Dict
 from typing import List
 from typing import Literal
+from typing import Optional
+from typing import Tuple
 from typing import Union
 
-from langchain_core.messages.base import BaseMessage
+
+from neuro_san.internals.messages.traced_message import TracedMessage
 
 
-class AgentFrameworkMessage(BaseMessage):
+class AgentFrameworkMessage(TracedMessage):
```

Comment:
Derive from TracedMessage now so these can be better displayed within LangSmith.